### ISSUE TITLE
On the bar, the toolbar icon will flash constantly.

### ISSUE THREAD

**Issue by suwat-netprom** [flutter/flutter-intellij#6467]
***What steps will reproduce the issue?***
1\. install plugin flutter
2\. open emulator
3\.run flutter on emulator
4\.close ide
5\.open another language project

***What is the expected result?***

toolbar icon should be display normally

**Spec**

IntelliJ IDEA 2022.3 (Ultimate Edition)

Build #IU-223.7571.182, built on November 29, 2022

macOS 13.1

GC: G1 Young Generation, G1 Old Generation

Memory: 1024M

Cores: 16

Non-Bundled Plugins:

&nbsp;&nbsp;&nbsp;&nbsp;org.intellij.plugins.hcl (223.7571.175)

&nbsp;&nbsp;&nbsp;&nbsp;Dart (223.7571.203)

&nbsp;&nbsp;&nbsp;&nbsp;pro.bashsupport (3.0.4.223)

&nbsp;&nbsp;&nbsp;&nbsp;com.itcodebox.leewyatt.notebooks.id (1.38)

&nbsp;&nbsp;&nbsp;&nbsp;io.flutter (71.1.6)

&nbsp;&nbsp;&nbsp;&nbsp;Pythonid (223.7571.182)

&nbsp;&nbsp;&nbsp;&nbsp;aws.toolkit (1.57-223)

&nbsp;&nbsp;&nbsp;&nbsp;Kotlin: 223-1.7.20-release-201-IJ7571.182

![Screen Recording 2565-12-16 at 10 02 17](https://user-images.githubusercontent.com/14134992/208129871-dbabe4d0-ac96-44ef-82c3-1c7a2932bb43.gif)

**Comment by alexander-doroshko** [comment]
I see the same.
![2022-12-16 16 42 31](https://user-images.githubusercontent.com/4499789/208135644-d298db5c-947c-45e5-a067-6ba310a61e46.gif)
macOS, IntelliJ IDEA 2022.3.1 RC (Build #IU-223.8214.27)
Flutter plugin 71.1.6

I simply opened SDK project to browse files (https://github.com/dart-lang/sdk/). This is not a Flutter project. I haven't enabled Dart/Flutter support - Dart SDK and Flutter SDK fields are empty in Settings.

**Comment by ksilz** [comment]
Same here in the old UI in a Java project:

https://user-images.githubusercontent.com/1061209/208138296-42b5e425-d013-4282-99af-f3a4f89d04a9.mp4



IntelliJ IDEA 2022.3 (Ultimate Edition)
Build #IU-223.7571.182, built on November 29, 2022
Runtime version: 17.0.5+1-b653.14 aarch64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 12.6.2
GC: G1 Young Generation, G1 Old Generation
Memory: 2048M
Cores: 10
Metal Rendering is ON
Non-Bundled Plugins:
    com.github.copilot (1.1.37.2188)
    lermitage.intellij.extra.icons (2022.1.14)
    bundled-intellij-idea-help (223.7571.200)
    net.seesharpsoft.intellij.plugins.csv (3.0.2-223)
    Dart (223.7571.203)
    Gitflow-Fix (0.7.11)
    io.flutter (71.1.6)
    zielu.gittoolbox (213.10.3)

Kotlin: 223-1.7.20-release-201-IJ7571.182

**Comment by stevemessick** [comment]
This flashing looks very annoying. We need another point release ASAP.

cc @jacob314

**Comment by stevemessick** [comment]
@alexander-doroshko Debugging has become very difficult, due to https://github.com/JetBrains/gradle-intellij-plugin/issues/1246

**Comment by GrigoriyKostyuk** [comment]
Same problem after plugin update/new install on old and new UI

IntelliJ IDEA 2022.3 (Ultimate Edition)
Build #IU-223.7571.182, built on November 29, 2022
macOS 13.1
GC: G1 Young Generation, G1 Old Generation
Memory: 2048M
Cores: 10
Metal Rendering is ON
Non-Bundled Plugins:
    com.jetbrains.edu (2022.12-2022.3-161)
    com.intellij.ideolog (203.0.30.0)
    ir.msdehghan.plugins.ansible (0.92)
     Dart (223.7571.203)
    io.flutter (71.1.6)

**Comment by ksilz** [comment]
Is there another way to select the Flutter device in IntelliJ? It seems I can only globally change the toolbar: Once I remove the Flutter device selector, it's also gone in my Flutter project.

**Comment by stevemessick** [comment]
@ksilz Sorry, but no. Until we can get a point release out you should probably use an earlier version of the Flutter plugin.

I've identified the problem. We were advised to add the device selector to another tool bar component, by [JetBrains](https://youtrack.jetbrains.com/issue/IDEA-298614/newUI-breaks-Flutter-plugin-device-selector#focus=Comments-27-6723690.0-0). That's what causes the flashing, I believe. Removing it solves the flashing problem. It also makes the new UI unusable.

My current plan is to build a new plugin that does not have the device selector in the new UI. We have another problem in progress (no embedded DevTools) so it will probably be early next week before the next release is ready.

**Comment by ksilz** [comment]
@stevemessick Thank you for the update! It ain't that bad &mdash; you don't have to fix it before Xmas for me.

**Comment by stevemessick** [comment]
@ksilz I should have given it more time. After re-opening a Java project the device selector started flashing. I have not found what causes it to appear, but the Flutter plugin makes it disappear. I currently have no idea what's going on.

**Comment by stevemessick** [comment]
I made some progress. The device selector widget is being made visible by IntelliJ, not the Flutter plugin. The plugin turns it off. One option, I suppose is to stop making it invisible, and just leave it visible all the time in all projects. Not the most appealing option, though.

https://youtrack.jetbrains.com/issue/MP-5019/IntelliJ-2022.3-causes-custom-toolbar-widget-to-flash

I managed to set a breakpoint where it is being made visible. Just so I don't lose it, this is the condition for stopping at the breakpoint:
```java
(getIcon() instanceof com.intellij.openapi.util.IconLoader.CachedImageIcon) && "/icons/phone.png".equals(((com.intellij.openapi.util.IconLoader.CachedImageIcon)getIcon()).getOriginalPath())
```
Here's the stack trace. The `Presentation` instance in the top frame is that of the device selector and the `visible` argument is true. 

<img width="700" alt="Screen Shot 2022-12-16 at 4 03 21 PM" src="https://user-images.githubusercontent.com/8518285/208211115-e05d62ef-7f7f-4d4b-9402-e8fbb6a561d1.png">
<img width="692" alt="Screen Shot 2022-12-16 at 4 04 08 PM" src="https://user-images.githubusercontent.com/8518285/208211132-f6dd02c1-5ef8-434c-932c-bbc3a4fa4633.png">

**Comment by stevemessick** [comment]
This is sort of fixed in 71.2. The UI framework makes the widget visible, even though it wasn't supposed to be. We can't change that so we made the widget visible in non-Flutter projects. At least it stops the flashing, which caused usability issues with the new UI.

**Comment by ksilz** [comment]
@stevemessick I upgraded to the Flutter plugin 71.2.6. I still see an empty device selector in my Java project, though it doesn't flicker anymore. And I just got this error.

IntelliJ IDEA `2022.3` • Flutter plugin `io.flutter 71.2.6` • Dart plugin `223.7571.203`

No Flutter sdk configured.

## Exception

null

```
com.intellij.diagnostic.PluginException: Empty menu item text for DeviceSelectorAction@ProjectViewPopup (io.flutter.actions.DeviceSelectorAction). The default action text must be specified in plugin.xml or its class constructor [Plugin: io.flutter]
	at com.intellij.diagnostic.PluginProblemReporterImpl.createPluginExceptionByClass(PluginProblemReporterImpl.java:23)
	at com.intellij.diagnostic.PluginException.createByClass(PluginException.java:83)
	at com.intellij.openapi.actionSystem.impl.Utils.reportEmptyTextMenuItem(Utils.java:504)
	at com.intellij.openapi.actionSystem.impl.Utils.fillMenuInner(Utils.java:427)
	at com.intellij.openapi.actionSystem.impl.Utils.fillMenu(Utils.java:355)
	at com.intellij.openapi.actionSystem.impl.ActionPopupMenuImpl$MyMenu.updateChildren(ActionPopupMenuImpl.java:177)
	at com.intellij.openapi.actionSystem.impl.ActionPopupMenuImpl$MyMenu.show(ActionPopupMenuImpl.java:129)
	at com.intellij.ui.PopupHandler$2.invokePopup(PopupHandler.java:142)
	at com.intellij.ui.PopupHandler.mousePressed(PopupHandler.java:47)
	at java.desktop/java.awt.AWTEventMulticaster.mousePressed(AWTEventMulticaster.java:288)
	at java.desktop/java.awt.AWTEventMulticaster.mousePressed(AWTEventMulticaster.java:287)
	at java.desktop/java.awt.Component.processMouseEvent(Component.java:6645)
	at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3392)
	at com.intellij.ui.treeStructure.Tree.processMouseEvent(Tree.java:424)
	at com.intellij.ide.dnd.aware.DnDAwareTree.processMouseEvent(DnDAwareTree.java:44)
	at java.desktop/java.awt.Component.processEvent(Component.java:6413)
	at java.desktop/java.awt.Container.processEvent(Container.java:2266)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5022)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4854)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4948)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4572)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4516)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2310)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2802)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4854)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:781)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:730)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:724)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:754)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:752)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:751)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:909)
	at com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.java:831)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:753)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$5(IdeEventQueue.java:437)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:774)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:436)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:113)
	at com.intellij.ide.IdeEventQueue.performActivity(IdeEventQueue.java:615)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:434)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:813)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:480)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
```
